backend/src/graphql/resolver/UserResolver.test.ts

```diff
@@ -1578,39 +1693,83 @@ describe('UserResolver', () => {
+            await mutate({
+              mutation: login,
+              variables: { email: 'peter@lustig.de', password: 'Aa12345_' },
+            })
+          })
+
+          afterAll(async () => {
+            await cleanDB()
+            resetToken()
           })
 
           describe('user gets new role', () => {
             describe('to admin', () => {
+              it('returns admin-rolename', async () => {
+                const result = await mutate({
+                  mutation: setUserRole,
+                  variables: { userId: user.id, role: ROLE_NAMES.ROLE_NAME_ADMIN },
+                })
+                expect(result).toEqual(
+                  expect.objectContaining({
+                    data: {
+                      setUserRole: ROLE_NAMES.ROLE_NAME_ADMIN,
+                    },
+                  }),
+                )
+              })
+
+              it('stores the ADMIN_USER_ROLE_SET event in the database', async () => {
+                const userContact = await UserContact.findOneOrFail({
+                  where: { email: 'bibi@bloxberg.de' },
+                  relations: ['user'],
+                })
+                const adminContact = await UserContact.findOneOrFail({
+                  where: { email: 'peter@lustig.de' },
+                  relations: ['user'],
+                })
+                await expect(DbEvent.find()).resolves.toContainEqual(
+                  expect.objectContaining({
+                    type: EventType.ADMIN_USER_ROLE_SET,
+                    affectedUserId: userContact.user.id,
+                    actingUserId: adminContact.user.id,
+                  }),
+                )
+              })
+            })
+
+            describe('to moderator', () => {
               it('returns date string', async () => {
                 const result = await mutate({
                   mutation: setUserRole,
-                  variables: { userId: user.id, isAdmin: true },
+                  variables: { userId: user.id, role: ROLE_NAMES.ROLE_NAME_MODERATOR },
                 })
                 expect(result).toEqual(
                   expect.objectContaining({
                     data: {
-                      setUserRole: expect.any(String),
+                      setUserRole: ROLE_NAMES.ROLE_NAME_MODERATOR,
                     },
                   }),
                 )
                 expect(new Date(result.data.setUserRole)).toEqual(expect.any(Date))
               })
 
               it('stores the ADMIN_USER_ROLE_SET event in the database', async () => {
-                const userConatct = await UserContact.findOneOrFail({
+                const userContact = await UserContact.findOneOrFail({
                   where: { email: 'bibi@bloxberg.de' },
                   relations: ['user'],
                 })
-                const adminConatct = await UserContact.findOneOrFail({
+                const adminContact = await UserContact.findOneOrFail({
```

Comment:
not needed. I know, this is not your code, but let us improve this.

Reply:
but why not ensure the event-writing?